src/hidimstat/_utils/scenario.py

```diff
@@ -220,134 +174,108 @@ def multivariate_simulation(
+    shuffle : bool, default=False
+        Whether to shuffle features randomly.
+    seed : int or None, default=None
         Random seed for reproducibility.
 
     Returns
     -------
     X : ndarray of shape (n_samples, n_features)
-        Design matrix with Toeplitz correlation structure.
-    Y : ndarray of shape (n_samples, n_times)
-        Target matrix Y = X @ beta + noise.
-    beta : ndarray of shape (n_features, n_times)
-        Parameter matrix with first support_size rows equal to 1.
-    noise : ndarray of shape (n_samples, n_times)
-        Temporally correlated Gaussian noise matrix.
-    """
-
-    rng = np.random.default_rng(seed)
-
-    X = np.zeros((n_samples, n_features))
-    X[:, 0] = rng.standard_normal(n_samples)
-
-    for i in np.arange(1, n_features):
-        rand_vector = ((1 - rho_data**2) ** 0.5) * rng.standard_normal(n_samples)
-        X[:, i] = rho_data * X[:, i - 1] + rand_vector
-
-    if shuffle:
-        rng.shuffle(X.T)
-
-    beta = np.zeros((n_features, n_times))
-    beta[0:support_size, :] = 1.0
-
-    noise = np.zeros((n_samples, n_times))
-    noise[:, 0] = rng.standard_normal(n_samples)
-
-    for i in range(1, n_times):
-        rand_vector = ((1 - rho_noise**2) ** 0.5) * rng.standard_normal(n_samples)
-        noise[:, i] = rho_noise * noise[:, i - 1] + rand_vector
-
-    noise = sigma * noise
-
-    Y = np.dot(X, beta) + noise
-
-    return X, Y, beta, noise
-
-
-def multivariate_1D_simulation_AR(
-    n_samples, n_features, rho=0.25, snr=2.0, sparsity=0.06, sigma=1.0, seed=None
-):
-    """
-    Function to simulate data follow an autoregressive structure with Toeplitz
-    covariance matrix
-
-    Parameters
-    ----------
-    n_samples : int
-        number of observations
-    n_features : int
-        number of variables
-    sparsity : float
-        ratio of number of variables with non-zero coefficients over total
-        coefficients
-    rho : float
-        Level of correlation between neighboring features.
-    effect : float
-        signal magnitude, value of non-null coefficients
-    seed : None or Int
-        random seed for generator
-
-    Returns
-    -------
-    X : ndarray, shape (n_samples, n_features)
-        Design matrix resulted from simulation
-    y : ndarray, shape (n_samples, )
-        Response vector resulted from simulation
-    beta_true : ndarray, shape (n_samples, )
-        Vector of true coefficient value
-    non_zero : ndarray, shape (n_samples, )
-        Vector of non zero coefficients index
+        Design matrix with Toeplitz covariance structure.
+    y : ndarray of shape (n_samples,) or (n_samples, n_times)
+        Target vector/matrix, generated as y = X @ beta + noise.
+    beta_true : ndarray of shape (n_features,) or (n_features, n_times)
+        True coefficient vector/matrix with support_size non-zero elements.
+    non_zero : ndarray
+        Indices of non-zero coefficients in beta_true.
+    noise_factor : float
```

Comment:
I think we should consistently use `snr` in the API instead of `noise_mag`, `noise_factor`  ...
It may add one line of code, but it improves clarity and uniformity.

Reply:
Why do you mean?
The noise_mag is the coefficient that adds it to the noise to have the right ratio and snr is the signal noise ratio.
For me, these two elements are quite different.

Reply:
I think these two are redundant. You define the `noise_mag` as a function of the `snr` ( L272/275).

Reply:
But if you have the `noise_mag` and the noisy signal, you can get the `snr`; similarly, if you have `snr`, noisy signal, you can get the `noise_mag`. 
My point is that having a consistent parameter controlling the noise level makes the code easier to understand and facilitates the API's use.

Reply:
> if you have `snr`, noisy signal, you can get the `noise_mag`.

For me, it's not possible because you don't have the signal amplitude. 
`noise_mag` is not a controlling variable, it's a result.

Reply:
Yes, but is it needed somewhere in the codebase. If not, I don't think it should be returned.

Reply:
I used it to compute noise covariance for testing reid.

Reply:
I don't really find a need to have a single function. Do it only if it is easy and simplifies the code base.

Reply:
I try to remove noise_mag but I realise that it's not possible to calculate back because I don't have access to the norm of eps.
The only way to have access is to remove the parameters `sigma_noise` or `snr`.

Reply:
@bthirion Do you think that the function of the generation of data is good?